zenodo/modules/pages/utils.py

```diff
@@ -66,6 +70,33 @@ def format_request_email_body(context):
     template = current_app.config['PAGES_EMAIL_BODY_TEMPLATE'],
     return render_template_to_string(template, context)
 
+
+def check_attachment_size(attachments):
+    """Function to check size of attachment within limits.
+
+    :param attachments: Attachments files.
+    :returns: Boolean.
+    """
+    if check_attachment_size(attachments) == 0:
+        return False
+    if attachments:
+        if 'Content-Length' in request.headers:
+            file_size = int(request.headers.get('Content-Length'))
+            if file_size > current_app.config['PAGES_ATTACHMENT_MAX_SIZE']:
+                return False
+        else:
+            size = 0
+            for upload in attachments:
+                upload.seek(current_app.config['PAGES_ATTACHMENT_MAX_SIZE'] -
+                            size)
+                if upload.read(1) == '':
+                    return False
+
+                upload.seek(0)
+                size += upload.read()
```

Comment:
I believe calling `read` at this point reads the actual bytes of the file and doesn't return the number of bytes. You are probably looking for something like `upload.seek(0); size += len(upload)`. You should test anyways with some large files to see what happens.